homeassistant/components/sensibo/climate.py

```diff
@@ -194,6 +192,7 @@ async def async_set_temperature(self, **kwargs) -> None:
             else:
                 return
 
+        assert temperature is not None
```

Comment:
It is not recommended to base use `assert` in production code unless a typing confuse mypy and the variable can't be `None`